admin/RunExport

```diff
@@ -50,6 +49,16 @@ chown "$FTP_USER:$FTP_GROUP" \
     "$FTP_DATA_DIR"/replication \
     "$FTP_DATA_DIR"/replication/{daily,weekly}
 
+# Create temporary directory under the same volume as the final
+# destination so that moving files does not require copying those.
+# If we're not doing a full export, just use default tmp instead.
+if [ "$WITH_FULL_EXPORT" ]
+then
+    make_temp_dir "$FTP_DATA_DIR"/fullexport/
```

Comment:
If I understand correctly this is a volume, then the data is copied somehow to final destination, right?
Can you explain the whole process?

https://github.com/metabrainz/musicbrainz-server/blob/771f3cc3362b42fdd1e364cf0a6ee136f35a7de9/admin/config.default.sh#L5
https://github.com/metabrainz/docker-server-configs/blob/bc51522124eb0f7dc6dd83ba676a996b93374230/scripts/services.sh#L1197
https://github.com/metabrainz/docker-server-configs/blob/bc51522124eb0f7dc6dd83ba676a996b93374230/scripts/services.sh#L315

Reply:
Right. This patches makes `make_temp_dir` to create a temporary directory in this volume (instead of in /tmp).

1. It dumps the database to files (under the temporary directory): https://github.com/metabrainz/musicbrainz-server/blob/v-2021-11-15/admin/RunExport#L53-L61
2. It copies these files to the backup destination (in another volume): https://github.com/metabrainz/musicbrainz-server/blob/v-2021-11-15/admin/RunExport#L65-L70
3. After removing files for private tables, it moves the rest to the final destination (in the same volume): https://github.com/metabrainz/musicbrainz-server/blob/v-2021-11-15/admin/RunExport#L77-L80
4. After organizing files and deleting old exports, it pushes changes to the `fullexport` directory to the FTP server: https://github.com/metabrainz/musicbrainz-server/blob/v-2021-11-15/admin/RunExport#L95

Note: “Full export” is the main database dump, but the code has been generalized to work with other kind of dumps (sample data, json dumps, incremental json dumps, search index dumps).